### PR TITLE
Added content to regular expression sections

### PR DESCRIPTION
**Pull request by theseion**

### CODE REVIEW

File: content/development/contribution_guidelines.md [hunk 1]
```diff
@@ -109,17 +109,107 @@ The older style of representing a backslash using the character class `[\\\\]` m
...
+- positive lookbehind (e.g. `(?<=regex)`)
+- negative lookbehind (e.g. `(?<!regex)`)
+- named capture groups (e.g. `(?P<name>regex)`)
+- backreferences (e.g. `\1`)
+- named backreferences (e.g. `(?P=name)`)
+- conditionals (e.g. `(?(regex)then|else)`)
+- recursive calls to capture groups (e.g. `(?1)`)
 
-TODO
+This list is not exhaustive but covers the most important points.
```
Comment (fzipi) [inline content/development/contribution_guidelines.md:212]: What about adding a list to upstream like https://github.com/google/re2/wiki/Syntax ?

Comment (theseion) [reply content/development/contribution_guidelines.md:212]: Good idea.

**Review by RedXanadu — COMMENTED** [review]
@theseion This is absolute **gold**! Thank you so much for writing this content.

I've made some minor changes in #26, if you're happy with those?

**Comment by RedXanadu** [comment]
Modifications discussed and accepted.

Now merging. (Thanks again @theseion!)

**Comment by theseion** [comment]
Thank **you** for spearheading the documentation effort! It's really important.

**Pull request merged**